### ISSUE TITLE
Comparision of GUID and Text results in Runtime Error (Compile Time would be better)

### ISSUE THREAD

**Issue by pri-kise** [microsoft/AL#6785]
**1. Describe the bug**
A GUID cannot be compared with Text, but i doens't results in a compile error.

**2. To Reproduce**

1. Create some Code with the following in it.
2. 

``` AL
    local procedure DummyGuidCheck()
    var
        DummyGuid: Guid;
        DummyText: Text;
    begin
        if DummyGuid <> DummyText then;
    end;
```

**3. Expected behavior**
I ecpect to receiv a Compiler Error instead of  a Runtime Error

**4. Actual behavior**
Right now the Code above is compiling but results in a runtime error, since text and guid cannot be compared.

**5. Versions:**

- AL Language: `9.0.519128` (Preview Release)
- Visual Studio Code:
Version: 1.60.2 (user setup)
Commit: 7f6ab5485bbc008386c4386d08766667e155244e
Date: 2021-09-22T12:00:31.514Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19042

- Business Central: Version: W1 20.0 (Plattform 20.0.30192.0 + Anwendung 20.0.30934.0)

### Final Checklist

Please remember to do the following:

* [X] Search the issue repository to ensure you are reporting a new issue

* [X] Reproduce the issue after disabling all extensions except the AL Language extension

* [X] Simplify your code around the issue to better isolate the problem

**Comment by dnpb** [comment]
What is the mysterious runtime error? Does it just say that Guid and Text cannot be compared, indicating _at all_? If the reality is that a well-formed Text can be compared with a Guid, the error should be clear about that, instead of implying it's never possible.

I guess it makes sense to be able to compare Guid with Text, as how else would one easily express an arbitrary GUID in code - but then I'd expect an error like 'Cannot compare Guid to an empty string' or 'The text is not in a valid GUID format' - etc.

**Comment by pri-kise** [comment]
@Jose-agg Thanks for your answer.
 Mabye a codecop info message, that informs the developer to be careful about the comparision and try to use a `GUID` variable together with an evaluate statement could be an simple temporary solution until there is enough time for more complex solutions?

**Comment by dnpb** [comment]
Agreed, comparing with a literal should be a compile-time error, but in the meantime an info/warning that ideally a `Guid` type would be used instead, would be preferable and avoid accidental production problems like I'm seeing just now. :-(
